moulinette/core.py

```diff
@@ -573,7 +573,7 @@ def acquire(self):
                     self._lock()
                     break
 
-            if (time.time() - start_time) > self.timeout:
+            if self.timeout is not None and (time.time() - start_time) > self.timeout:
```

Comment:
For me, it makes more sens to mean "don't wait at all" but this point could be discussed as I have no strong opinion on it.